Packs/CiscoUmbrellaReporting/Integrations/CiscoUmbrellaReporting/CiscoUmbrellaReporting.py

```diff
@@ -0,0 +1,1138 @@
+            "Date & Time": timestamp_string
+        }
+        activity_list.append(new)
+    headers = activity_list[0] if activity_list else {}
+    headers = list(headers.keys())
+    markdown = tableToMarkdown(title, activity_list, headers=headers, removeNull=True)
+    return markdown
+
+
+@logger
+def activity_ip_lookup_to_markdown(results: List[Dict], title: str) -> \
+        str:
+    activity_list = []
+    for activity in results:
+        category = [label.get("label") for label in activity.get(
+            'categories', [])]
+        identity = [label.get("label") for label in activity.get(
+            'identities', [])]
+        timestamp = activity.get("timestamp", 0)
+        timestamp_string = datetime.utcfromtimestamp(
+            timestamp / 1000.0).strftime(DATE_TIME_FORMAT)
+        new = {
+            "Identity": ",".join(identity),
+            "Destination IP": activity.get("destinationip", ''),
+            "Source IP": activity.get("sourceip", ''),
+            "Source Port": activity.get("sourceport", ''),
+            "Destination Port": activity.get("destinationport", ''),
+            "Categories": ",".join(category),
+            "Type": activity.get("type", ''),
+            "Action": activity.get("verdict", ''),
+            "Date & Time": timestamp_string
+        }
+        activity_list.append(new)
+    headers = activity_list[0] if activity_list else {}
+    headers = list(headers.keys())
+    markdown = tableToMarkdown(title, activity_list, headers=headers, removeNull=True)
+    return markdown
+
+
+@logger
+def activity_amp_lookup_to_markdown(results: List[Dict], title: str) -> \
+        str:
+    activity_list = []
+    for activity in results:
+        timestamp = activity.get("timestamp", 0)
+        timestamp_string = datetime.utcfromtimestamp(
+            timestamp / 1000.0).strftime(DATE_TIME_FORMAT)
+        new = {
+            "First Seen": activity.get("firstseenat", ''),
+            "Disposition": activity.get("disposition", ''),
+            "Score": activity.get("score", ''),
+            "Host Name": activity.get("hostname", ''),
+            "Malware": activity.get("malwarename", ''),
+            "SHA256": activity.get("sha256", ''),
+            "Date & Time": timestamp_string
+        }
+        activity_list.append(new)
+    headers = activity_list[0] if activity_list else {}
+    headers = list(headers.keys())
+    markdown = tableToMarkdown(title, activity_list, headers=headers, removeNull=True)
+    return markdown
+
+
+@logger
+def pagination(args: Dict):
+    page = args.pop('page', None)
+    if page and int(page) <= 0:
+        raise DemistoException(PAGE_NUMBER_ERROR_MSG)
+    page = 0 if not page else (int(page) - 1)
+    page_size = int(args.pop('page_size', 0))
+    args["limit"] = page_size if page_size > 0 else args.get('limit', 50)
+    args['offset'] = page * page_size
+    return args
+
+
+@logger
+def get_param(args: Dict) -> Dict:
```

Comment:
done

Reply:
@RamboV - I think you didn't understand me well. 
I was trying to say that we usually don't edit the original demito.args() dictionary -  because editing it might make it harder to debug the code in the future.
I didn't mean you to copy the demito.args() to a new variable in the main function... my intention was to extract the needed arguments from the original args dictionary, and create a new dictionary containing the relevant arguments (with their right keys) to transfer to cisco umbrella API. 

I'll try to explain in detail what I want you to change (if it isn't clear please contact me and we will set up a meeting):

1.  Please remove this line from the main function - `copy_args = copy.deepcopy(args)` - this is unnecessary. 
2. Please change the name of the `get_param` function to `create_cisco_umbrella_args`. 
3. Please improve the function's docstring - I think that we need to add something like - "This function creates a dictionary of the arguments sent to the Cisco Umbrella API based on the demisto.args()."
4. In the function itself, please generate an empty dictionary (call it 'cisco_umbrella_args' maybe). 
5. For each argument of the args (demisto.args()) dict: a. extract it from the args dict. b. add it to the cisco_umbrella_args dict with the relevant key.
6. If the argument shouldn't pass to the API call (for example "summary_type" or "traffic_type") - don't add it to the cisco_umbrella_args dict (that will replace the calls to the args.pop() you used in the commands). Remove the args.pop() from the relevant commands. 
7. Return the cisco_umbrella_args.
8. When calling the function - ` params = create_cisco_umbrella_args(limit, offset, args)` please change the `params `variable name to 'cisco_umbrella_args' or 'new_args'.
9. I think you can also move the following block of code inside the function - instead of calling it from each command:
```
    if sha256 := args.get('sha256'):
        check_valid_indicator_value("sha256", sha256)
    if ip := args.get('ip'):
        check_valid_indicator_value("ip", ip)
    if domains := args.get("domains"):
        check_valid_indicator_value('domains', domains)
```

Please contact me if something is unclear.

Reply:
@ShacharKidor Done

Reply:
Great job!